### DISCUSSION TITLE
Error when running "make -C ulib/ml"

### DISCUSSION THREAD

**Discussion by Daniel-Cumming** [FStarLang/FStar#2625] [Q&A] [answered]
Hello, I am trying to follow the directions from the github to run the codegen OCaml and execute F*. But in the early step of running "make -C ulib/ml", I run into this error:
```
...
File "ulib/tactics_ml/fstarlib_leftovers/FStar_Seq_Base.ml", line 1:
Error: /home/zncv/Applications/fstar/bin/fstar-compiler-lib/prims.cmi
       is not a compiled interface for this version of OCaml.
It seems to be for a newer version of OCaml.
Command exited with code 2.
Compilation unsuccessful after building 8 targets (0 cached) in 00:00:00.
```
My version of OCaml
```
$ ocaml --version
The OCaml toplevel, version 4.11.1
```
How can you tell which version of OCaml is required to process the binary?

**Comment by tahina-pro** [comment, marked as answer]
Are you using the F* binary release? If so, then you have to use the same version of OCaml as the one F* was compiled with, i.e. OCaml 4.12, as stated in https://github.com/FStarLang/FStar/blob/master/INSTALL.md#using-a-binary-package-to-compile-ocaml-code-generated-by-f

**Reply by Daniel-Cumming** [reply]
Oh thank you, I looked in this file for the answer but somehow missed this.
